__tests__/decorator.ts

```diff
@@ -58,4 +58,18 @@ describe('@WithUntilDestroyed decorator', () => {
 
     expect(callback).not.toHaveBeenCalled();
   });
+
```

Comment:
Please add more tests for any possible scenario. Check the existing tests. Each time there is a new bug, and I don't have time to check the code. Thanks.